numpy/core/src/umath/loops.c.src

```diff
@@ -845,6 +846,109 @@ NPY_NO_EXPORT NPY_GCC_OPT_3 void
 
 /**end repeat1**/
 
+/*
+ * Arithmetic bit shift operations.
+ *
+ * Intel hardware masks bit shift values, so large shifts wrap around
+ * and can produce surprising results. The special handling ensures that
+ * behavior is independent of compiler or hardware.
+ * TODO: We could implement consistent behavior for negative shifts,
+ *       which is undefined in C.
+ */
+
+#define LEFT_SHIFT_OP                               \
+    do {                                            \
+        if (NPY_LIKELY(in2 < sizeof(@type@) * 8)) { \
```

Comment:
A cleaner version modulo signedness might be
```
#define LEFT_SHIFT = do {*out = NPY_LIKELY(in2 < NPY_BITSOF_@TYPE@) ? in1 << in2 : 0} while(0)
```
Although I don't think we actually need the `do ... while(0)` bit here.